arroyo-worker/src/operators/functions/regexp.rs

```diff
@@ -0,0 +1,51 @@
+use regex::Regex;
+
+pub fn regexp_match(argument: String, regex: String) -> Vec<String> {
+    let re = Regex::new(&regex).unwrap();
+    match re.captures(argument.as_str()) {
+        Some(caps) => caps
+            .iter()
+            .enumerate()
+            .filter_map(|(i, cap)| {
+                if i != 0 {
+                    cap.map(|m| m.as_str().to_owned())
+                } else {
+                    None
+                }
+            })
+            .collect(),
+        None => vec![],
+    }
+}
+
+pub fn regexp_replace(argument: String, regex: String, replacement: String) -> String {
+    let re = Regex::new(&regex).unwrap();
+    let result = re.replace_all(&argument, replacement.as_str());
+    result.into_owned()
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use std::error::Error;
+
+    #[test]
+    pub fn test_regexp_match_is_correct() -> Result<(), Box<dyn Error>> {
```

Comment:
In general you don't need to have return types on your tests. Either it panics or it passes.

Reply:
Fixed with https://github.com/ArroyoSystems/arroyo/pull/122/commits/9889c37a41d5d3ae6f678c500f5cbcacf754ad9e 

I am normally more used to work with functional errors rather than panics, that's why I naturally leaned on using results